generators/server/templates/src/main/java/package/web/rest/_AccountResource.java

```diff
@@ -423,53 +440,58 @@ public void changePassword(@RequestBody String password) {
-           persistentTokenRepository.findByUser(u).stream()
-               .filter(persistentToken -> StringUtils.equals(persistentToken.getSeries(), decodedSeries))<% if (databaseType === 'sql' || databaseType === 'mongodb') { %>
-               .findAny().ifPresent(t -> persistentTokenRepository.delete(decodedSeries)));<% } else { %>
-               .findAny().ifPresent(persistentTokenRepository::delete));<% } %>
-   }<% } %>
-
-   /**
+    @DeleteMapping("/account/sessions/{series}")
+    @Timed
+    public void invalidateSession(@PathVariable String series) throws UnsupportedEncodingException {
+        String decodedSeries = URLDecoder.decode(series, "UTF-8");
+        userRepository.findOneByLogin(SecurityUtils.getCurrentUserLogin()).ifPresent(u ->
+            persistentTokenRepository.findByUser(u).stream()
+                .filter(persistentToken -> StringUtils.equals(persistentToken.getSeries(), decodedSeries))<% if (databaseType === 'sql' || databaseType === 'mongodb') { %>
+                .findAny().ifPresent(t -> persistentTokenRepository.delete(decodedSeries)));<% } else { %>
+                .findAny().ifPresent(persistentTokenRepository::delete));<% } %>
+    }<% } %>
+
+    /**
     * POST   /account/reset-password/init : Send an email to reset the password of the user
     *
     * @param mail the mail of the user
     * @throws EmailNotFoundException 400 (Bad Request) if the email address is not registered
     */
-   @PostMapping(path = "/account/reset-password/init")
-   @Timed
-   public void requestPasswordReset(@RequestBody String mail) {
-       mailService.sendPasswordResetMail(
-           userService.requestPasswordReset(mail)
-               .orElseThrow(EmailNotFoundException::new)
-       );
-   }
+    @PostMapping(path = "/account/reset-password/init")
+    @Timed
+    public void requestPasswordReset(@RequestBody String mail) {
+        Optional<User> user = userService.requestPasswordReset(mail);
+        if (!user.isPresent()) {
+            throw new EmailNotFoundException();
+        }
+        mailService.sendPasswordResetMail(user.get());
+    }
 
-   /**
+    /**
     * POST   /account/reset-password/finish : Finish to reset the password of the user
     *
     * @param keyAndPassword the generated key and the new password
     * @throws InvalidPasswordException 400 (Bad Request) if the password is incorrect
     * @throws RuntimeException 500 (Internal Server Error) if the password could not be reset
     */
-   @PostMapping(path = "/account/reset-password/finish")
-   @Timed
-   public void finishPasswordReset(@RequestBody KeyAndPasswordVM keyAndPassword) {
-       if (!checkPasswordLength(keyAndPassword.getNewPassword())) {
-           throw new InvalidPasswordException();
-       }
-       userService.completePasswordReset(keyAndPassword.getNewPassword(), keyAndPassword.getKey())
-           .orElseThrow(RuntimeException::new);
```

Comment:
`.orElseThrow(() -> new DefaultException("No user was found for this reset key."));` would be better

Reply:
.orElseThrow(() -> ProblemBuilder.withTitle("No user was found for this reset key.").build());